lib/ansible/module_utils/common/parameters.py

```diff
@@ -673,7 +673,9 @@ def _validate_argument_values(argument_spec, parameters, options_context=None, e
                         if len(overlap) == 1:
                             (parameters[param],) = overlap
 
-                    if parameters[param] not in choices:
+                    # Error if param isn't in choices, and it's not the default value
+                    # default may be None and None may not be in choices
+                    if parameters[param] not in choices and parameters[param] != spec.get('default', object()):
```

Comment:
this will affect it if a user is explicitly passing the default as an option (iirc we had issue, will try to find)

Reply:
I've removed this change.